doc/Route_priority_and_TTL.md

```diff
@@ -0,0 +1,131 @@
+
+![](./images/route_priority_ttl/priority_message_pump_flowchart.jpg)
+
+The message pump will not attempt to aggregate messages of different priorities into a single batch.  This is to guarantee that we’ll only need to process one type of priority at a time, which greatly simplifies the checkpointing requirements.
+To process the messages from a given priority, we’ll need a corresponding instance of Checkpointer to keep track of failures and retries.  To accomplish this, the executor will maintain a map of priorities to FSMs, the latter of which will now contain the checkpointer specific to its queue.
+
+### Handling TTL
+As mentioned above, the TTL specific to a route will now be available when a message gets stored.  This TTL value will be a new property on the stored MessageRef object, so that it can be referenced when the message pump processes the message.
+
+As the message pump fetches messages out of the store, the MessageStore::GetNext() implementation will check the message enqueue timestamp against the TTL timespan on the message for expiration.  If the message has no TTL, then global TTL will be used.
+
+If the message is expired, it simply gets skipped over, and GetNext() keeps on iterating until it fills the requested batch size or the store is empty.  Then in the background, the existing message cleanup task will periodically delete these expired messages.
+
+### Live deployment manifest changes
+A new deployment manifest can come down with routing updates while there are still pending messages in the store.  In this case, since we’ve effectively snapshotted the previous priority and TTL values into the store, we won’t do anything to change behavior of the pending messages.  Any new message that comes in after the manifest changes are applied will be enqueued with the new config.
+
+### Data Migration
+Since storage changes are additive, data migration is not needed.  We have fallback legacy values for everything we’re adding, so if the newer logic encounters an old message, it should be able to handle it as normal.
+
+### Security and Performance
+None of the threat boundaries are changing with this feature, so there are no new security concerns.
+
+There should also be no noticeable impact to performance, as we’re modifying the order in which messages are sent by categorizing messages in storage.  The added I/O will be trivial, and the fundamentals of message forwarding is preserved.  
```

Comment:
Can you add a section on the limitations? For instance to explicitely document that we restrict the number of priorities to not exceed 10, etc. Thanks!